plaso/parsers/shared/unified_logging/activity.py

```diff
@@ -0,0 +1,243 @@
+    formatter_flags = ffh.FormatFlags(tracev3, flags, data, offset)
+    offset = formatter_flags.offset
+
+    if flags & constants.PRIVATE_STRING_RANGE:
+      logger.error('Activity with Private String Range unsupported')
+      return
+
+    # If there's data...
+    if tracepoint.data_size - offset >= 6:
+      data_meta = tracev3.ReadStructureFromByteStream(
+          data[offset:], offset,
+          tracev3.GetDataTypeMap('tracev3_firehose_tracepoint_data'))
+      offset += 2
+
+      logger.debug(
+          'After activity data: Unknown {0:d} // Number of Items {1:d}'.format(
+              data_meta.unknown1, data_meta.num_items))
+      try:
+        (log_data, deferred_data_items,
+          offset) = tracev3.ReadItems(data_meta, data, offset)
+      except errors.ParseError as exception:
+        logger.error('Unable to parse data items: {0!s}'.format(exception))
+        return
+
+      if flags & constants.HAS_CONTEXT_DATA != 0:
+        logger.error('Backtrace data in Activity log chunk unsupported')
+        return
+
+      if flags & constants.HAS_DATA_REF:
+        logger.error('Activity log chunk with Data Ref unsupported')
+        return
+
+      for item in deferred_data_items:
+        if item[2] == 0:
+          result = ''
+        elif item[0] in constants.FIREHOSE_ITEM_PRIVATE_STRING_TYPES:
+          if not private_string:
+            logger.error('Trying to read from empty Private String')
+            return
+          try:
+            result = tracev3.ReadStructureFromByteStream(
+                private_string[item[1]:], 0, tracev3.GetDataTypeMap('cstring'))
+            logger.debug('End result: {0:s}'.format(result))
+          except errors.ParseError:
+            result = ''  # Private
+        else:
+          if item[0] in constants.FIREHOSE_ITEM_STRING_ARBITRARY_DATA_TYPES:
+            result = data[offset + item[1]:offset + item[1] + item[2]]
+          elif item[0] == constants.FIREHOSE_ITEM_STRING_BASE64_TYPE:
+            result = base64.encodebytes(data[offset + item[1]:offset + item[1] +
+                                             item[2]]).strip()
+          else:
+            result = tracev3.ReadStructureFromByteStream(
+                data[offset + item[1]:], 0, tracev3.GetDataTypeMap('cstring'))
+            logger.debug('End result: {0:s}'.format(result))
+        log_data.insert(item[3], (item[0], item[2], result))
+
+    if formatter_flags.shared_cache or formatter_flags.large_shared_cache != 0:
+      extra_offset_value_result = tracepoint.format_string_location
+      if formatter_flags.large_offset_data != 0:
+        if (
+            formatter_flags.large_offset_data
+            != formatter_flags.large_shared_cache / 2
+            and not formatter_flags.shared_cache
+        ):
+          # Recovery ?
+          formatter_flags.large_offset_data = int(
+              formatter_flags.large_shared_cache / 2
+          )
+          extra_offset_value = '{0:X}{1:08x}'.format(
```

Comment:
Why this approach? why not just calculate the value as an integer instead of casting it to a string and then back to an integer?